accepts invalid code with undefined type

``` bash
$ scalac-2.12.0 t.scala
t.scala:2: error: not found: type test
  def test (x : test) : Int
                ^
one error found
$ 
$ dotc -version

Dotty compiler version 0.01 -- (c) 2013 LAMP/EPFL
$ 
$ dotc t.scala
Warning: mocking up superclass for module class test
$ 
$ cat t.scala
trait Foo {
  def test (x : test) : Int
} 
$ 
```

--- a/src/dotty/tools/dotc/core/Symbols.scala
+++ b/src/dotty/tools/dotc/core/Symbols.scala
@@ -428,7 +428,7 @@ object Symbols {
     final def entered(implicit ctx: Context): this.type = {
       assert(this.owner.isClass, s"symbol ($this) entered the scope of non-class owner ${this.owner}") // !!! DEBUG
       this.owner.asClass.enter(this)
-      if (this is Module) this.owner.asClass.enter(this.moduleClass)
+      if (this.is(Module, butNot = Package)) this.owner.asClass.enter(this.moduleClass)
       this
     }
 

--- a/src/dotty/tools/dotc/typer/Typer.scala
+++ b/src/dotty/tools/dotc/typer/Typer.scala
@@ -126,9 +126,8 @@ class Typer extends Namer with TypeAssigner with Applications with Implicits wit
      *  case x :: xs in class List would return the :: method).
      */
     def qualifies(denot: Denotation): Boolean =
-      reallyExists(denot) &&
-        !(name.isTypeName && denot.symbol.is(Package)) &&
-        !(pt.isInstanceOf[UnapplySelectionProto] &&
+      reallyExists(denot) && !(
+        pt.isInstanceOf[UnapplySelectionProto] &&
           (denot.symbol is (Method, butNot = Accessor)))
 
     /** Find the denotation of enclosing `name` in given context `ctx`.

--- a/src/dotty/tools/dotc/typer/Typer.scala
+++ b/src/dotty/tools/dotc/typer/Typer.scala
@@ -126,9 +126,10 @@ class Typer extends Namer with TypeAssigner with Applications with Implicits wit
      *  case x :: xs in class List would return the :: method).
      */
     def qualifies(denot: Denotation): Boolean =
-      reallyExists(denot) && !(
-         pt.isInstanceOf[UnapplySelectionProto] &&
-         (denot.symbol is (Method, butNot = Accessor)))
+      reallyExists(denot) &&
+        !(name.isTypeName && denot.symbol.is(Package)) &&
+        !(pt.isInstanceOf[UnapplySelectionProto] &&
+          (denot.symbol is (Method, butNot = Accessor)))
 
     /** Find the denotation of enclosing `name` in given context `ctx`.
      *  @param previous    A denotation that was found in a more deeply nested scope,

--- a/tests/neg/i1643.scala
+++ b/tests/neg/i1643.scala
@@ -1,4 +1,4 @@
-trait T extends Array {
+trait T extends Array { // error
   def t1(as: String*): Array[String] = { varargs1(as: _*) } // error
   def t2(as: String*): Array[String] = { super.varargs1(as: _*) } // error
 }

--- a/tests/neg/i1701.scala
+++ b/tests/neg/i1701.scala
@@ -1,0 +1,3 @@
+trait Foo {
+  def test (x : test) : Int
+}

--- a/tests/neg/i1701.scala
+++ b/tests/neg/i1701.scala
@@ -1,3 +1,5 @@
 trait Foo {
-  def test (x : test) : Int
+  def test (x : test) : Int // error: not found test
+  def f(x : src) : Int // error
+  def g(x : src.project) : Int // error
 }

